bluemira/power_cycle/tools.py

```diff
@@ -31,3 +37,65 @@ def create_axes(ax=None):
     if ax is None:
         _, ax = plt.subplots()
     return ax
+
+
+@nb.jit
+def unique_domain(x: np.ndarray, epsilon: np.number = DEFAULT_EPSILON):
+    """
+    Ensure x has only unique values to make (Domain: x -> Image: y) a function.
+
+    Epsilon must be small enough so that consecutive values can be considered
+    equal within the context/scale in which the function is defined.
+
+    Nudge forward each non-unique element in x by (N * epsilon), given N times,
+    after the first appearance, that value has appeared in x before.
+    """
+    n_points = len(x)
+    new_x = [x[0]]
+    if n_points > 1:
```

Comment:
n points is only used in if statement, no need for a variable.
```suggestion
    new_x = [x[0]]
    if  len(x) > 1:
```

Reply:
it's legacy, I used it more before the last modifications you suggested